src/cors-proxy/init.lua

```diff
@@ -47,31 +47,52 @@ local api_docs_headers = {
   'X-Apidocs-Path', 'X-Apidocs-Url', 'X-Apidocs-Query', 'X-Apidocs-Method'
 }
 
-function _M:rewrite()
-  local url = resty_url.split(ngx.var.http_x_apidocs_url)
-  local upstream = {
-    server = url[4],
-    port = url[5] or resty_url.default_port(url[1]),
-    host = url[4],
-    path = ngx.var.http_x_apidocs_path or ngx.var.uri,
-    args = ngx.var.http_x_apidocs_query or '',
-    method = METHODS[ngx.var.http_x_apidocs_method],
-  }
-
-  ngx.ctx.upstream = upstream
-  ngx.req.set_header('Host', upstream.host)
-  ngx.var.proxy_scheme = url[1]
-
-  for i=1,#api_docs_headers do
-    ngx.req.clear_header(api_docs_headers[i])
-  end
+local function set_cors_headers()
+  local origin = ngx.var.http_origin
 
-  if upstream.method then
-    ngx.req.set_method(upstream.method)
-  end
+  if not origin then return end
+
+  ngx.header['Access-Control-Allow-Headers'] = ngx.var.http_access_control_request_headers
+  ngx.header['Access-Control-Allow-Methods'] = ngx.var.http_access_control_request_method
+  ngx.header['Access-Control-Allow-Origin'] = origin
+  ngx.header['Access-Control-Allow-Credentials'] = 'true'
+end
 
-  ngx.var.proxy_path = upstream.path
-  ngx.req.set_uri_args(upstream.args)
+local function cors_preflight_response()
+  set_cors_headers()
+  ngx.status = 204
+  ngx.exit(ngx.status)
+end
+
+function _M:rewrite()
+  if cors_preflight() then
```

Comment:
Have you tried this? This can't work.

Reply:
Yes sorry about that, I was checking it and I mixed up copying the stuff.. gimme 5 minutes